From dmonakhov@sw.ru Sat Mar  3 05:10:55 2007
From: Dmitriy Monakhov <dmonakhov@sw.ru>
Date: Sat, 03 Mar 2007 16:11:21 +0300
Subject: kobject: new_device->kref wasn't putted in kobject_move()
To: Dmitriy Monakhov <dmonakhov@sw.ru>
Cc: linux-kernel@vger.kernel.org, gregkh@suse.de
Message-ID: <87lkieqxom.fsf@sw.ru>

From: Dmitriy Monakhov <dmonakhov@sw.ru>

[PATCH] kobject: new_device->kref wasn't putted after error in kobject_move()

If error happen we jump to "out" label, in this case new_device not yet
became the parent but it wasn't putted.

Signed-off-by: Monakhov Dmitriy <dmonakhov@openvz.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 lib/kobject.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/lib/kobject.c
+++ b/lib/kobject.c
@@ -385,9 +385,11 @@ int kobject_move(struct kobject *kobj, s
 		goto out;
 	old_parent = kobj->parent;
 	kobj->parent = new_parent;
+	new_parent = NULL;
 	kobject_put(old_parent);
 	kobject_uevent_env(kobj, KOBJ_MOVE, envp);
 out:
+	kobject_put(new_parent);
 	kobject_put(kobj);
 	kfree(devpath_string);
 	kfree(devpath);
